engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/JobPreparationActor.scala

```diff
@@ -9,85 +9,56 @@ import wdl4s.expression.WdlStandardLibraryFunctions
+  case class BackendJobPreparationSucceeded(jobDescriptor: BackendJobDescriptor, props: Props) extends JobPreparationActorResponse
+  case class BackendJobPreparationFailed(jobKey: JobKey, throwable: Throwable) extends JobPreparationActorResponse
 
   def props(executionData: WorkflowExecutionActorData,
             jobKey: BackendJobDescriptorKey,
             factory: BackendLifecycleActorFactory,
-            configDescriptor: BackendConfigurationDescriptor) = Props(new JobPreparationActor(executionData, jobKey, factory, configDescriptor))
+            configDescriptor: BackendConfigurationDescriptor) = {
+    Props(new JobPreparationActor(executionData, jobKey, factory, configDescriptor)).withDispatcher("akka.dispatchers.slow-actor-dispatcher")
+  }
 }
 
 case class JobPreparationActor(executionData: WorkflowExecutionActorData,
                                jobKey: BackendJobDescriptorKey,
                                factory: BackendLifecycleActorFactory,
-                               configDescriptor: BackendConfigurationDescriptor) extends LoggingFSM[JobPreparationActorState, Unit] with WdlLookup {
+                               configDescriptor: BackendConfigurationDescriptor) extends Actor with WdlLookup {
 
   override val workflowDescriptor: EngineWorkflowDescriptor = executionData.workflowDescriptor
   override val executionStore: ExecutionStore = executionData.executionStore
   override val outputStore: OutputStore = executionData.outputStore
   override val expressionLanguageFunctions = executionData.expressionLanguageFunctions
 
-  startWith(Idle, Unit)
-
-  when(Idle) {
-    case Event(Start, _) =>
-      resolveAndEvaluate(jobKey, expressionLanguageFunctions) onComplete {
-        case Success(inputs) => self ! EvaluatedInputs(inputs)
-        case Failure(f) =>
-          context.parent ! BackendJobPreparationFailed(jobKey, f)
-          context stop self
+  override def receive = {
+    case Start =>
+      val response = resolveAndEvaluate(jobKey, expressionLanguageFunctions) flatMap { prepareJobExecutionActor } match {
+        case Success(m) => m
+        case Failure(f) => BackendJobPreparationFailed(jobKey, f)
       }
-      goto(EvaluatingInputs)
-  }
 
-  when(EvaluatingInputs) {
-    case Event(EvaluatedInputs(inputs), _) =>
-      val jobDescriptor = BackendJobDescriptor(workflowDescriptor.backendDescriptor, jobKey, inputs)
-      val config = BackendConfigurationDescriptor(configDescriptor.backendConfig, configDescriptor.globalConfig)
-
-      Future(factory.jobExecutionActorProps(jobDescriptor, config)) onComplete {
-        case Success(props) =>
-          context.parent ! BackendJobPreparationSucceeded(jobDescriptor, props)
-          context stop self
-        case Failure(f) =>
-          context.parent ! BackendJobPreparationFailed(jobKey, f)
-          context stop self
-      }
-      goto(BuildingProps)
+      context.parent ! response
+      context stop self
   }
```

Comment:
I don't remember if this was intentional or not. I'm guessing that it's because I lost the builtin log and kinda doubt we'll ever use these unhandled log messages anyways - but I'll add it back in.